pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -221,6 +221,10 @@ func validateNetworksMatchInterfaces(field *k8sfield.Path, spec *v1.VirtualMachi
 
 	vifMQ = spec.Domain.Devices.NetworkInterfaceMultiQueue
 	isVirtioNicRequested = false
+	podNetworkAutoattached := spec.Domain.Devices.AutoattachPodInterface == nil || *spec.Domain.Devices.AutoattachPodInterface
+	if podNetworkAutoattached && len(spec.Networks) == 0 && len(spec.Domain.Devices.Interfaces) == 0 {
```

Comment:
I think we attach the pod network even if you request another net/if.

Reply:
Please see here, only if nothing exists (condition + comment)
https://github.com/kubevirt/kubevirt/blob/67fc3608dcfe26c9c37d891d3944ae08e270e05e/pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator.go#L212

Reply:
So correct me if I am wrong but here you will already have net/if and fail on the condition.

Reply:
Also maybe the documentation around `autoattachPodInterface` should be adjusted.

Reply:
> So correct me if I am wrong but here you will already have net/if and fail on the condition.

We check if there are no interfaces, and if auto attach is enabled, if both conditions met, it means that a virtio interface will be added, so `isVirtioNicRequested` is set to true.
This boolean is then checked as part of the admitter, if MQ is enabled, it expects isVirtioNicRequested to be true.

The mentioned function `setDefaultNetworkInterface` is something bit different, it is the function that adds upon needs the auto attached interface.

> Also maybe the documentation around autoattachPodInterface should be adjusted.

You mean this comment ?
If so, it is fine, it override only if no interfaces exists (unless you mean something else ?)
https://github.com/kubevirt/kubevirt/blob/67fc3608dcfe26c9c37d891d3944ae08e270e05e/pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator.go#L211

Reply:
> > So correct me if I am wrong but here you will already have net/if and fail on the condition.
> 
> We check if there are no interfaces, and if auto attach is enabled, if both conditions met, it means that a virtio interface will be added, so `isVirtioNicRequested` is set to true. This boolean is then checked as part of the admitter, if MQ is enabled, it expects isVirtioNicRequested to be true.

The ordering is mutating webhook -> validating webhook. Is it that validating webhook doesn't get the defaulted object? (It is a long time I have looked)

> 
> The mentioned function `setDefaultNetworkInterface` is something bit different, it is the function that adds upon needs the auto attached interface.
> 
> > Also maybe the documentation around autoattachPodInterface should be adjusted.
> 
> You mean this comment ? If so, it is fine, it override only if no interfaces exists (unless you mean something else ?)
> 
> https://github.com/kubevirt/kubevirt/blob/67fc3608dcfe26c9c37d891d3944ae08e270e05e/pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator.go#L211


I meant https://github.com/kubevirt/kubevirt/blob/67fc3608dcfe26c9c37d891d3944ae08e270e05e/staging/src/kubevirt.io/api/core/v1/schema.go#L415. What you pointed is code, not documentation. (well sort of... best documentation is code :) )

Reply:
> The ordering is mutating webhook -> validating webhook. Is it that validating webhook doesn't get the defaulted object? (It is a long time I have looked)

Right, but this validating webhook runs also for the VM, not just for the VMI, so there we can get a machine without interfaces, and if we will add an interface depends on the `AutoattachPodInterface`

 > https://github.com/kubevirt/kubevirt/blob/67fc3608dcfe26c9c37d891d3944ae08e270e05e/staging/src/kubevirt.io/api/core/v1/schema.go#L415

What change you suggest here ?
(not sure if it is part of this PR, but lets see first please what is the suggestion)

Reply:
Thanks, somehow I didn't make the connection. 

Right, I didn't mean we need to do it here.
What do you mean by
>(also isn't it too detailed to add this info there?)

Reply:
That if you meant the doc should be changed to
`// Whether to attach a pod network interface (in case no network and interfaces exists already). Defaults to true.`
It might be too detailed ? (but it is arguable)

Reply:
I would say it creates false expectations (that it will be attached even when there are other net/if) without it.